daily-challenges/2024-09/19/hunghoang3110.spec.ts

```diff
@@ -0,0 +1,42 @@
+// Javascript
+let productList = {}
+function addProduct (name, price) {
+    productList[name] = price;
+}
+
+function removeProduct (name) {
+    delete productList[name]
+}
+
+function caculateTotal () {
+    let total = 0;
+    for (let [product, value] of Object.entries(productList)) {
+        total += value;
+    }
+    return total
+}
+
+// Playwright
+import {test, expect} from '@playwright/test'
+
+test('play puzzle', async ({page}) => {
+
+    // Navigate to Puzzle drag and drop game page
+    await page.goto('https://material.playwrightvn.com/');
+    await expect(page).toHaveURL(/.*playwrightvn.com/);
+    await page.getByRole('link', { name: /Puzzle drag and drop game/ }).click();
+    await expect(page).toHaveURL(/.*drag-and-drop/);
+
+    //Drag and drop puzzel
+    await page.locator('#piece-1').dragTo(page.locator('[data-piece="1"]'));
+    await page.locator('#piece-2').dragTo(page.locator('[data-piece="2"]'));
+    await page.locator('#piece-3').dragTo(page.locator('[data-piece="3"]'));
+    await page.locator('#piece-4').dragTo(page.locator('[data-piece="4"]'));
+
+    // Verify
+    page.on('dialog', dialog => {
```

Comment:
Logic handle dialog/alert cần đẩy lên trước khi sự kiện dialog/alert diễn ra

Reply:
Copied that